osu.Game/Graphics/UserInterface/HoverClickSounds.cs

```diff
@@ -34,10 +35,19 @@ public HoverClickSounds(HoverSampleSet sampleSet = HoverSampleSet.Normal, MouseB
             this.buttons = buttons ?? new[] { MouseButton.Left };
         }
 
+        private ScheduledDelegate playDelegate;
+
         protected override bool OnClick(ClickEvent e)
         {
+            playDelegate?.Cancel();
+
             if (buttons.Contains(e.Button) && Contains(e.ScreenSpaceMousePosition))
-                sampleClick?.Play();
+            {
+                if (DebounceTime <= 0)
+                    sampleClick?.Play();
+                else
+                    playDelegate = Scheduler.AddDelayed(() => sampleClick?.Play(), DebounceTime);
```

Comment:
are you sure we need debounce on click? i don't think anyone can click this fast.

Reply:
I thought it might make more sense for click sounds to also be debounced, since `HoverClickSounds` derives `HoverSounds`.

Reply:
I don't think it's necessary, and would probably feel wrong if a click didn't make a sound.

Reply:
I've split click debounce from hover debounce. Click debounce will be disabled by default, but I still think it should be there for the symmetry.

Reply:
Safe to say we won't use it in osu!, as it goes against the UX we are aiming for.